src/Stratis.Bitcoin.Features.SignalR/SignalRFeature.cs

```diff
@@ -106,11 +108,24 @@ public static partial class IFullNodeBuilderExtensions
                     {
                         services.AddSingleton<IEventsSubscriptionService, EventSubscriptionService>();
                         services.AddSingleton<EventsHub>();
-                        services.AddSingleton<IClientEventBroadcaster, PeerStatisticsClientBroadcaster>();
-                        services.AddSingleton<IClientEventBroadcaster, StakingBroadcaster>();
                         services.AddSingleton(fullNodeBuilder);
                         services.AddSingleton(options);
                         services.AddSingleton<SignalRSettings>();
+
+                        if (null != options.ClientEventBroadcasters)
+                        {
+                            foreach (Type eventBroadcaster in options.ClientEventBroadcasters)
+                            {
+                                if (typeof(IClientEventBroadcaster).IsAssignableFrom(eventBroadcaster))
+                                {
+                                    services.AddSingleton(typeof(IClientEventBroadcaster), eventBroadcaster);
+                                }
+                                else
+                                {
+                                    Console.WriteLine($"Warning {eventBroadcaster.Name} is not of type {typeof(IClientEventBroadcaster).Name}");
```

Comment:
Use `logger` instead of `Console.WriteLine`

Reply:
This is a static method on the builder, I couldn't find an example with logging and how to get the logger into here, do you know of an example ?

Reply:
yeah, with Microsoft logging it is not simple. Leave it for now